boost.cmake

```diff
@@ -193,18 +193,23 @@ MACRO(SEARCH_FOR_BOOST)
 ENDMACRO(SEARCH_FOR_BOOST)
 
 #.rst:
-# .. command:: TARGET_LINK_BOOST_PYTHON (TARGET)
+# .. command:: TARGET_LINK_BOOST_PYTHON (TARGET <PRIVATE|PUBLIC|INTERFACE>)
 #
 #   Link target againt boost_python library.
 #
-#   :TARGET: is either a library or an executable
+#   :target: is either a library or an executable
+#   :private,public,interface: The PUBLIC, PRIVATE and INTERFACE keywords can be used to specify both the link dependencies and the link interface.
 #
 #   On darwin systems, boost_python is not linked against any python library.
 #   This linkage is resolved at execution time via the python interpreter.
 #   We then need to stipulate that boost_python has unresolved symbols at compile time for a library target.
 #   Otherwise, for executables we need to link to a specific version of python.
 #
 MACRO(TARGET_LINK_BOOST_PYTHON target)
+  IF(${ARGC} GREATER 1)
+    SET(PUBLIC_KEYWORD ${ARGV1})
```

Comment:
Won't this mess up with the current usage of PUBLIC_KEYWORD ?

Reply:
I don't think this should be manager globally, but rather locally.

Reply:
I agree with you.
But at the moment, there are still places where the global variable is accessed. We should modify them so that they don't use the global variable anymore.